src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1872,4 +1879,128 @@ sym2.ind (λ v w, is_bridge_iff_adj_and_forall_cycle_not_mem) e
 
 end bridge_edges
 
+section induced
+
+/-! ### Connectivity of induced graphs -/
+
+lemma induce_singleton_connected (v : V) :
+  (G.induce {v}).connected :=
+begin
+  rw connected_iff,
+  refine ⟨_, by simp⟩,
+  rintro ⟨x, hx⟩ ⟨y, hy⟩,
+  rw set.mem_singleton_iff at hx hy,
+  subst_vars,
+end
+
+lemma induce_union_connected {s t : set V}
+  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) (sintert : (s ∩ t).nonempty ) :
+  (G.induce $ s ∪ t).connected :=
+begin
+  obtain ⟨u, hus, hut⟩ := sintert,
+  rw connected_iff_basepoint,
+  use ⟨u, or.inl hus⟩,
+  rintro ⟨v, hv|hv⟩,
+  { obtain ⟨p⟩ := (sconn ⟨u, hus⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_left : (G.induce s) →g (G.induce $ s ⊔ t))⟩, },
+  { obtain ⟨p⟩ := (tconn ⟨u, hut⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_right : (G.induce t) →g (G.induce $ s ⊔ t))⟩, },
+end
+
+lemma induce_pair_connected_of_adj {u v : V} (huv : G.adj u v) :
+  (G.induce {u, v}).connected :=
+begin
+  rw connected_iff,
+  refine ⟨_, by simp⟩,
+  rintro ⟨x, hx⟩ ⟨y, hy⟩,
+  simp only [set.mem_insert_iff, set.mem_singleton_iff] at hx hy,
+  obtain rfl|rfl := hx; obtain rfl|rfl := hy;
+    refl <|> { refine ⟨walk.cons _ walk.nil⟩, simp [huv, huv.symm] }
+end
+
+lemma induce_connected_adj_union {s t : set V}
+  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) {v w} (hv : v ∈ s) (hw : w ∈ t)
+  (a : G.adj v w) : (G.induce $ s ∪ t).connected :=
+begin
+  have : s ∪ t = s ∪ {v, w} ∪ t, by
+  { rw [set.union_comm s {v, w}, set.union_assoc], symmetry,
+    apply set.union_eq_self_of_subset_left,
+    simp only [set.insert_subset, set.singleton_subset_iff, hv, hw, set.mem_union, true_or,
+               or_true, and_self], },
+  rw this,
+  refine induce_union_connected
+    (induce_union_connected sconn (induce_pair_connected_of_adj a) _) tconn _,
+  { refine ⟨v, hv, _⟩, simp, },
+  { refine ⟨w, or.inr _, hw⟩, simp, }
+end
+
+lemma induce_walk_support_connected [decidable_eq V] :
+  ∀ {u v : V} (p : G.walk u v), (G.induce $ (p.support.to_finset : set V)).connected
```

Comment:
This is how you're supposed to convert a list into a set:
```suggestion
  ∀ {u v : V} (p : G.walk u v), (G.induce {v | v ∈ p.support}).connected
```

That said, I don't think this lemma is formulated correctly. We should use `p.to_subgraph`, use that it is connected, have a separate lemma that `p.to_subgraph` is a subgraph of `(⊤ : G.subgraph).induce {v | v ∈ p.support}`, and then combine these with some of the support lemmas I wrote to prove this one, if it's needed.

https://github.com/leanprover-community/mathlib/blob/kmill_hamiltonian/src/hamiltonian.lean#L1199